### PR TITLE
redoing input check for mixed models

### LINKED ISSUE #867: Mixed Models - dependent variable checks cannot deal with NAs
as reported here: https://github.com/jasp-stats/jasp-issues/issues/843

### PR DESCRIPTION
**Pull request by FBartos**
fixes: https://github.com/jasp-stats/jasp-issues/issues/867

changes:
- removes rows with NA/NaN/Inf directly after loading the dataset
- creates the note about omitted variables based on this

### CODE REVIEW

File: JASP-Engine/JASP/R/MixedModelsCommon.R [hunk 1]
```diff
@@ -97,50 +97,71 @@
...
+  # check and use only the variables that actually used for modeling
+  used_variables <- .v(c(
+    options$dependentVariable,
+    if(type %in% c("GLMM", "BGLMM")) if(options$dependentVariableAggregation != "") options$dependentVariableAggregation,
+    unique(unlist(options$fixedEffects)),
+    if(length(options$randomVariables) != 0) options$randomVariables
+  ))
+  dataset <- dataset[,used_variables]
+  
+  # omit NAs/NaN/Infs and store the number of omitted observations
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/MixedModelsCommon.R:142]: There are some standard na.omit functions in R that might be useful. They are probably optimised for dropping parts of the data. @JohnnyDoorn might know more about this

Comment (FBartos) [reply JASP-Engine/JASP/R/MixedModelsCommon.R:142]: but `na.omit()` doesn't omit Inf

Comment (AlexanderLyNL) [reply JASP-Engine/JASP/R/MixedModelsCommon.R:142]: Yes that's very true, I'm just thinking about performance. For my analyses I actually don't remove `Inf`, but I catch it with .hasErrors and display an error.

Comment (FBartos) [reply JASP-Engine/JASP/R/MixedModelsCommon.R:142]: Yeah, that's the good point. I changed it to `na.omit()` and error for `Inf`

File: JASP-Engine/JASP/R/MixedModelsCommon.R [hunk 2]
```diff
@@ -97,50 +97,78 @@
...
-        ))
+        )
       }  
     }
   }
+
+  dataset <- data.frame(dataset)
+
+  # check and use only the variables that actually used for modeling
+  used_variables <- .v(c(
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/MixedModelsCommon.R:134]: This is some what hard to read. And no space after if, and no space after a comma.

File: JASP-Engine/JASP/R/MixedModelsCommon.R [hunk 3]
```diff
@@ -97,50 +97,78 @@
...
+    options$dependentVariable,
+    if(type %in% c("GLMM", "BGLMM")) if(options$dependentVariableAggregation != "") options$dependentVariableAggregation,
+    unique(unlist(options$fixedEffects)),
+    if(length(options$randomVariables) != 0) options$randomVariables
+  ))
+  dataset <- dataset[,used_variables]
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/MixedModelsCommon.R:140]: Here's the missing space after the comma

File: JASP-Engine/JASP/R/MixedModelsCommon.R [hunk 4]
```diff
@@ -97,50 +97,78 @@
...
+  
+  # omit NAs/NaN/Infs and store the number of omitted observations
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/MixedModelsCommon.R:142]: INF can be removed from the comment

**Review by AlexanderLyNL — COMMENTED** [review]
Overall it looks fine. Some style issues that you're suffering from though, but otherwise it's okay.

**Pull request merged**